sass/cd/cd-header/_cd-language-switcher.scss

```diff
@@ -6,123 +6,142 @@
   }
 }
 
-.cd-user-menu__dropdown-link {
-  display: block;
-  padding: 12px 0;
-  text-decoration: none;
+// Pipe separator.
+.cd-language-switcher__dropdown li a::before {
+  position: absolute;
+  top: 50%;
+  left: -1px;
+  width: 1px;
+  height: calc(var(--cd-global-header-height) / 3);
+  content: "";
+  transform: translateY(-50%);
+  background: var(--brand-grey);
+}
+// No pipe on first child.
+.cd-language-switcher__dropdown li:first-child a::before {
+  background: transparent;
+}
 
-  &.is-active {
-    position: relative;
-    font-weight: 700;
+// If there are menus in the Global Header
+// and the Language Switcher is displayed in full,
+// separate with pipes and margins.
+.cd-global-header__language-switcher + .cd-global-header__user-menu {
+  position: relative;
+  // Margin value matches the padding of the other buttons in the Global Header
+  // and takes into account the right padding from the last Language Item.
+  margin-inline-start: 0.5rem;
 
-    [dir="ltr"] & {
-      padding-right: 12px;
-    }
+  @include tablet {
+    margin-inline-start: 1rem;
+  }
+}
+// Pipe separator between Language Switcher and following menu.
+.cd-global-header__language-switcher + .cd-global-header__user-menu::before {
+  position: absolute;
+  top: 50%;
+  left: -1px;
+  width: 2px;
+  height: calc(var(--cd-global-header-height) / 3);
+  content: "";
+  transform: translateY(-50%);
+  background: var(--brand-grey);
+}
 
-    [dir="rtl"] & {
-      padding-left: 12px;
-    }
+// Adjust breakpoints for when the Language Switcher displays as a dropdown.
+// This depends on other elements present in the .cd-global-header__actions.
+//
+// Copy the following rules into the sub theme and override breakpoint value.
+@include max-width(sm) {
+  .js {
+    .cd-language-switcher__dropdown {
+      @include z-index(dropdown);
 
-    &::after {
+      display: block;
       position: absolute;
+      left: 0;
       right: 0;
-      width: 14px;
-      height: 14px;
-      content: "";
-      background-image: url("../img/icons/selected--highlighted.svg");
-      background-repeat: no-repeat;
-      background-position: 0 50%;
-      background-size: 14px 14px;
-
-      [dir="rtl"] & {
-        right: auto;
-        left: 0;
+      left: auto;
+      min-width: 125px;
+      padding: 12px 24px;
+      color: var(--cd-white);
+      background-color: var(--brand-primary);
+      box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.3);
+
+      a::before {
+        content: none;
       }
     }
+
+    // No horizontal separation when Language Switcher displays in a dropdown.
+    .cd-global-header__language-switcher + .cd-global-header__user-menu {
+      margin-inline-start: 0;
+    }
   }
 }
 
-.no-js .cd-user-menu__dropdown {
-  width: unset;
-  margin-left: 12px;
-  padding: 0;
-
-  li {
-    display: inline-block;
-    padding: 0 12px;
+// Display the full Language Switcher on > mobile.
+.cd-language-switcher {
+  @include mobile {
+    --dropdown: false;
```

Comment:
I've set the default breakpoint for when the Language Switcher changes to a dropdown as mobile ( < 576 ). This needs to be tested. Maybe < 768px is more realistic. This can be tested on a few sites